src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VisualBasicVSImports.cs

```diff
@@ -0,0 +1,187 @@
+            {
+                _threadingService.ExecuteSynchronously(async () =>
+                {
+                    using (var access = await _lockService.WriteLockAsync())
+                    {
+                        var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                        await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                        project.AddItem(importItemTypeName, bstrImport);
+                    }
+                });
+
+                OnImportAdded(bstrImport);
+            }
+            else
+            {
+                throw new ArgumentException(string.Format("{0} - Namespace is already imported", bstrImport), nameof(bstrImport));
+            }
+        }
+
+        public void Remove(object index)
+        {
+            bool intIndexPresent = index is int indexInt && _importsList.IsPresent(indexInt);
+            bool stringIndexPresent = index is string removeImport && _importsList.IsPresent((string)removeImport);
+            if (intIndexPresent || stringIndexPresent)
+            {
+                string importRemoved = null;
+                _threadingService.ExecuteSynchronously(async () =>
+                {
+                    using (var access = await _lockService.WriteLockAsync())
+                    {
+                        Microsoft.Build.Evaluation.ProjectItem importProjectItem = null;
+                        var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                        await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                        if (index is string removeImport1)
+                        {
+                            importProjectItem = project.GetItems(importItemTypeName)
+                                                       .First(i => string.Compare(removeImport1, i.EvaluatedInclude, StringComparison.OrdinalIgnoreCase) == 0);
+                        }
+                        else if (index is int indexInt1)
+                        {
+                            importProjectItem = project.GetItems(importItemTypeName)
+                                                       .OrderBy(i => i.EvaluatedInclude)
+                                                       .ElementAt((indexInt1 - 1));
+                        }
+                        else
+                        {
+                            // Cannot reach this point, since index has to be Int or String
+                            System.Diagnostics.Debug.Assert(false, $"Parameter {nameof(index)} is niether an int nor a string");
+                        }
+
+                        if (importProjectItem.IsImported)
+                        {
+                            throw new ArgumentException(string.Format(VisualBasicVSResources.ImportsFromTargetCannotBeDeleted, index.ToString()), nameof(index));
```

Comment:
Instead of throwing an exception, we should throw a dialog box showing this operation is not valid. Will open an issue.